File: backend/src/main/java/com/plugga/backend/dao/DeckCardDAO.java
package com.plugga.backend.dao;

import com.plugga.backend.entity.DeckCard;

import java.util.List;

public interface DeckCardDAO {
    public List<DeckCard> findAll();

    public DeckCard findById(int id);

    public void save(DeckCard deckCard);

    public void deleteById(int id);
}

File: package/src/main/kotlin/com/plugga/backend/dao/DeckCardDAO.kt
package com.plugga.backend.dao

import com.plugga.backend.entity.DeckCard

interface DeckCardDAO {
    fun findAll(): MutableList<DeckCard>

    fun findById(id: Int): DeckCard?

    fun findByDeckId(deckId: Int): MutableList<DeckCard>

    fun findByCardId(cardId: Int): MutableList<DeckCard>

    fun save(deckCard: DeckCard)

    fun deleteById(id: Int)
}
